src/lib/helpers/helpers.ts

```diff
@@ -0,0 +1,19 @@
+function distinct(array: any[], comparator?: (value1: any, value2: any) => number): any[] {
+	if (comparator) {
+		return distinctWithComparator(array, comparator);
+	} else {
+		return distinctWithoutComparator(array);
+	}
+}
+
+function distinctWithComparator(array: any[], comparator: (value1: any, value2: any) => number): any[] {
+	return array.filter((value, index, array) => index === 0 || comparator(value, array[index - 1]) !== 0);
+}
+
+function distinctWithoutComparator(array: any[]): any[] {
+	return array.filter((value, index, array) => index === 0 || value !== array[index - 1]);
```

Comment:
My only concern with these is they rely on the array being sorted, if I'm reading it correctly. We should probably rewrite this to either sort the items or to use a different method to make the array distinct.

Reply:
Yeah not a bad idea.

One method I found when searching was to filter and then compare the `indexOf(value)` to the current index. `indexOf` will always return the index of the *first* instance of the given value, so the comparison will be false for repeat entries.

Reply:
The only problem with that approach is that `indexOf()` only does `===` comparison, so we can't pass in a custom comparator. However, we can sort the list in the distinct function, which i have now done.